src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1077,8 +1077,11 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         FontInfo actualFont = { fontFace, 0, 10, { 0, fontHeight }, CP_UTF8, false };
         FontInfoDesired desiredFont = { actualFont };
 
-        const auto cols = settings.InitialCols();
-        const auto rows = settings.InitialRows();
+        // If the settings have negative or zero row or column counts, ignore those counts.
+        // (The lower TerminalCore layer also has upper bounds as well, but at this layer
+        //  we may eventually impose different ones depending on how many pixels we can address.)
+        const auto cols = std::max(settings.InitialCols(), 1);
+        const auto rows = std::max(settings.InitialRows(), 1);

```

Comment:
Note that regardless of this change, Terminal still crashes on startup if you have more than a few thousand rows in the `initialRows` setting, for reasons unrelated to this bug fix (since technically the GitHub issue is only about the `historySize` setting). I'll open a separate GitHub issue for that bug.

Reply:
opened: https://github.com/microsoft/terminal/issues/859